tests/plugins/release-monitor/index.js

```diff
@@ -27,105 +39,107 @@ const RELEASE_LABEL = "release";
- * @param {string} _ ignored param
- * @param {string} payload payload sent ot the API
- * @returns {undefined}
- * @private
+ * Mocks a pending status for a given status ID.
+ * @param {string} statusId The ID of the status to mock.
+ * @returns {void}
  */
-function assertSuccessStatusWithNoPendingRelease(_, payload) {
-    const data = JSON.parse(payload);
-
-    expect(JSON.parse(payload).state).toBe("success");
-    expect(data.description).toBe("No patch release is pending");
-    expect(data.target_url).toBe("");
+function mockStatusPending(statusId) {
+    fetchMock.mockGlobal().post({
+        url: `${API_ROOT}/repos/test/repo-test/statuses/${statusId}`,
+        body: {
+            state: "pending",
+            description: "A patch release is pending",
+            target_url: "https://github.com/test/repo-test/issues/1"
+        },
+        matchPartialBody: true
+    }, 200);
 }
 
 /**
- * Asserts that the state value is success
- * @param {string} _ ignored param
- * @param {string} payload payload sent ot the API
- * @returns {undefined}
- * @private
+ * Mocks a successful status update for a given status ID.
+ * @param {string} statusId The ID of the status to update.
+ * @returns {void}
  */
-function assertSuccessStatusWithPendingRelease(_, payload) {
-    const data = JSON.parse(payload);
-
-    expect(JSON.parse(payload).state).toBe("success");
-    expect(data.description).toBe("This change is semver-patch");
+function mockStatusSuccess(statusId) {
+    fetchMock.mockGlobal().post({
+        url: `${API_ROOT}/repos/test/repo-test/statuses/${statusId}`,
+        body: {
+            state: "success",
+            description: "This change is semver-patch"
+        },
+        matchPartialBody: true
+    }, 200);
 }
 
 describe("release-monitor", () => {
     let bot = null;
 
-    beforeAll(async () => {
-        bot = new Application({
-            id: "test",
-            cert: "test",
-            cache: {
-                wrap: () => Promise.resolve({ data: { token: "test" } })
-            },
-            app: () => "test"
+    beforeEach(() => {
+        bot = new Probot({
+            appId: 1,
+            githubToken: "test",
+            Octokit: ProbotOctokit.defaults(instanceOptions => ({
+                ...instanceOptions,
+                throttle: { enabled: false },
+                retry: { enabled: false }
+            }))
         });
-
-        const { paginate } = await bot.auth();
-
-        bot.auth = () => Object.assign(new GitHubApi(), { paginate });
         releaseMonitor(bot);
     });
 
     afterEach(() => {
-        nock.cleanAll();
+        fetchMock.unmockGlobal();
+        fetchMock.removeRoutes();
+        fetchMock.clearHistory();
     });
 
     describe("issue labeled", () => {
-        test("in post release phase then add appropriate status check to all PRs", async () => {
+        test.only("in post release phase then add appropriate status check to all PRs", async () => {
```

Comment:
This skips other tests in this file. Is it intentional?

Reply:
Oops, definitely not.